### PR TITLE
Reddit shortener

### PR DESCRIPTION
**Pull request by CtrlAltMech**
rshort module properly grabbing information and posting to IRC as well as the modified link in the description. This was done by creating a reddit_link method in the Shortener class and calling it from rshort. The shortener module will also not be triggered by reddit links anymore since that will be handled by rshort to place in the link description. Date and time of reddit post has also been added to description as well. First pull request on github, hope this worked out ok.

### CODE REVIEW

File: modules/rshort.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      
+
+    def handle(self, event):
+
+      url = re.search(self.r_pattern, event.line).group(0)
+      try: 
+        sub = reddit.submission(url='https://'+url)
+      except praw.exceptions.ClientException: # If the link is broken we should not do anything at all
+        return
+      sub_time = datetime.utcfromtimestamp(sub.created).strftime('%Y/%m/%d--%H:%M') # converts from POSIX timestamp to human readable
```
Comment (hlmtre) [inline modules/rshort.py]:
```suggestion
      sub_time = datetime.utcfromtimestamp(sub.created).strftime('%Y/%m/%d@%H:%M') # converts from POSIX timestamp to human readable
```

File: modules/rshort.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  successful_import = True
+except (ImportError, SystemError):
+  print("Warning: rshort module requires praw https://github.com/praw-dev/praw/")
+  successful_import = False
+except praw.exceptions.ClientException:
+  print('Warning: rshort requires the "praw.ini" file in the top pybot directory\
+  to be filled in with the client_id and client_secret.')
+  successful_import = False
+
+if successful_import is True:
```
Comment (hlmtre) [inline modules/rshort.py]:
```suggestion
if not successful_import:
  print("couldn't import praw or praw.ini not completed!")
  return
  
# unindent the rest of the file; we'd return from it first instead if we can't move on
```

**Review by hlmtre — CHANGES_REQUESTED** [review]
Looks pretty good but I'd like to verify it behaves like I think: shorten either way, but if it's a reddit link, also print reddit info.

Definitely also change to fail fast too.

**Comment by CtrlAltMech** [comment]
Will add these changes in along with the other changes I am making after doing some research. Thanks for the feedback.

**Review by hlmtre — APPROVED** [review]

**Pull request merged**